core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -435,9 +436,11 @@ private[kafka] abstract class ConsumerStageLogic[K, V, Out](
   // impl of Consumer.Control that is called from outside via materialized value
   override def isShutdown: Future[Done] = shutdownPromise.future
 
-  override def stop(): Unit = {
+  override def stop(): Future[Done] = {
     log.debug("Stop producing messages")
-    stopCallback.invoke(())
+    val result = Promise[Done]
```

Comment:
`result` will be completed in callback so every promise will be completed, but I agreed that calling `complete` multiple times smells.